docs/DeveloperGuide.md

```diff
@@ -177,6 +177,60 @@ Classes used by multiple components are in the `seedu.guest.commons` package.
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Add feature
+
+#### Implementation:
+
+* The `add` command takes in 6 compulsory fields (`Name`, `Phone`, `Email`, `Room Number`, `Date Range` and `Number of Guests`) and 1 optional field (`Request`) and is supported by the `AddCommandParser` that extracts out each of the fields from their respective prefixes.
+
+The following activity diagrams summarizes what happens when a user enters an `add` command.
+
+![AddActivityDiagram](images/AddActivityDiagram.png)
+
+#### Design Considerations:
+
+**Aspect: How to deal with duplicate entries**
+
+* **Alternative 1 (current choice):** Reject any entry with a duplicate name that is case-insensitive.
+    * Pros: Easy to implement.
+    * Cons: Will have issues with adding people of the same name to GuestBook.
+
+* **Alternative 2:** Allow duplicate entries to be added with no consequences.
+    * Pros: Allows for multiple guests of the same full name to be added.
+    * Cons: Easy to abuse the command intentionally (to break the program) or unintentionally (accidentally).
+
+Taking into consideration that in small hotels, the chances of getting two guests that have identical full names are
+incredibly unlikely, we decided to proceed with Alternative 1.
+
+**Aspect: Excluding `isRoomClean` in the `add` command**
+* As adding the guest will be done during check in, the room should by default be cleaned for the guest to immediately
+be able to enter their room. Hence, we chose to remove `isRoomClean` as it makes the `add` command unnecessarily longer.
+
+**Aspect: Excluding `Bill` in the `add` command**
+* As adding the guest will be done during check in, the guest should not have incurred any bill costs yet. Hence,
+we chose to remove `Bill` as it makes the `add` command unnecessarily longer.
+
+**Aspect: Making `Request` in the `add` command**
+* As adding the guest will be done during check in, the guest might not have any special requests to make for the room. Hence,
+we chose to make `Request` optional and default it as blank should it not be provided.
+
+### Edit feature
+
+#### Implementation:
+* The `edit` command takes in an INDEX indicating the index of the guests to edit in the current panel (starting from 1) and 8 optional fields (`Name`, `Phone`, `Email`, `Room Number`, `Date Range`, `Number of Guests`, `Room Clean` and `Request`) and is supported by the `EditCommandParser` that extracts out each of the fields from their respective prefixes.
```

Comment:
Likewise, should this be Room instead of Room Number?

Reply:
Agreed. Changed.